probe-rs/src/debug/variable.rs

```diff
@@ -655,214 +699,239 @@ impl Variable {
         self.value.is_valid()
     }
 
+    /// Format the variable.
     fn formatted_variable_value(
         &self,
-        variable_cache: &variable_cache::VariableCache,
+        variable_cache: &VariableCache,
         indentation: usize,
         show_name: bool,
-    ) -> String {
-        let line_feed = if indentation == 0 { "" } else { "\n" };
-        let line_start = format!("{}{:\t<indentation$}", line_feed, "");
+    ) -> Option<String> {
         let type_name = self.type_name();
 
         if !self.value.is_empty() {
-            // Use the supplied value or error message.
-            if show_name {
+            // This is the end of the recursion where we already have a scalar
+            // value for a variable and we can just move it up.
+            let line_start = line_indent_string(indentation);
+            return Some(if show_name {
```

Comment:
Too bad Rust doesn't have a `¿` operator to propagate the `Ok() / Some()` cases, we could write this fn as a linear series of `self.try_this_case()¿` calls.